src/middlewares/role-permissions.middleware.js

```diff
@@ -0,0 +1,50 @@
+const { User, Role, Permission } = require('@models/index');
+
+module.exports = function rolePermissions(permittedRoles,permittedPermissions) {
+
+  return async (request, response, next) => {
+    const { id } = request.payload;
+    const user = await User.findByPk(id,{
+      include: [
+        {
+          model: Role,
+          include: [
+            {
+              model: Permission,
+              as: 'permissions'
+            }
+          ]
+        }
+      ]
+    });
+
+    // Role validation
+    const haveRole = permittedRoles.includes(user.Role.name);
+
+    // Permission validation
+    let havePermission;
+    
+    /* eslint-disable array-callback-return */
+    /* eslint-disable no-restricted-syntax */
+    /* eslint-disable no-loop-func */
+    for (const iterator of permittedPermissions) {
+      user.Role.permissions.some((permission) => {
+        if(permission.description === iterator){
+          havePermission = true;
+        }
+      })
+    }
+    
+    if(user.Role.name === 'Superadmin') {
```

Comment:
why are you evaluating if is super admin after for permission?,
in case user is super admin always are going to execute for and in this case is unnecessary because is superadmin.